### ISSUE TITLE
[a11y] - IndexTable's `promotedBulkActions` links are unable to refocus after closing modal

### ISSUE THREAD

**Issue by aeperea** [Shopify/polaris-react-archive#9465]
## Issue summary

While TAB navigating, if you click on an promotedBulkAction button that opens a modal, and then you close the modal, the focus is not back in the promotedBulkAction button and it is lost. The prop promotedBulkAction buttons doesn't seem to be able to be wrapped around a ref to use as 'activator'.

## Expected behavior

Whenever the IndexTable's' promotedBulkAction button opens a modal and this is closed, the focus should be directed to the button that opened it 

## Steps to reproduce the problem

1. First step: Tab navigate to an Index Table that has actions that prompt a modal
2. Second step: Open the modal with keyboard navigation
3. Third step: Close the modal with keyboard navigation
4. Forth step: Where is the current focus?

Example in Shopify web: https://screenshot.click/14-37-5be43-f20qe.mp4

## Reduced test case

The best way to get your bug fixed is to provide a reduced test case. This [CodeSandbox template](https://codesandbox.io/s/q82mlq0m26) is a great starting point.

## Specifications

```bash
  System:
    OS: Linux 5.10 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (32) x64 Intel(R) Xeon(R) CPU @ 2.80GHz
    Memory: 98.85 GB / 251.90 GB
    Container: Yes
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.20.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.19.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/bin/watchman
  npmPackages:
    @shopify/polaris: 11.1.2 => 11.1.2
    react: ^18.2.0 => 18.2.0
    react-dom: ^18.2.0 => 18.2.0
```
